core/blockchain.go

```diff
@@ -340,8 +328,14 @@ func FinalizeBlockExecution(
 	}
 	if isMining {
 		newBlock, newTxs, newReceipt, err = engine.FinalizeAndAssemble(cc, header, ibs, txs, uncles, receipts, withdrawals, requests, chainReader, syscall, nil, logger)
```

Comment:
`FinalizeAndAssemble` and `FinalizeBlockExecution` should return requests, because we need to assign them to `current.Requests` in `SpawnMiningExecStage`